homeassistant/components/analytics/analytics.py

```diff
@@ -115,6 +123,43 @@ async def save_preferences(self, preferences: dict) -> None:
                 self.hass, self.preferences.get(ATTR_DIAGNOSTICS, False)
             )
 
+    async def _gather_configured_domains(self) -> Set:
+        """Return a set of configured_domains."""
+        configured_domains = set()
+        configured_domains.update(self.hass.config_entries.async_domains())
+
+        config_dict = await conf_util.async_hass_config_yaml(self.hass)
+
+        for domain in config_dict:
```

Comment:
This logic is flawed. We can't rely on this.

We support things like `light bla:` and we support platform config as both lists and dictionaries for integrations, but only if define `PLATFORM_SCHEMA` or if they deal with it themselves via `config.py` platform.